modules/plottinggl/processors/parallelcoordinates/parallelcoordinates.cpp

```diff
@@ -443,134 +326,69 @@ void ParallelCoordinates::process() {
     utilgl::activateAndClearTarget(outport_, ImageType::ColorPicking);
     utilgl::GlBoolState depthTest(GL_DEPTH_TEST, false);
 
-    buildTextCache(dims, enabledAxis);
-
-    extraMargins_ = extraMarginsHandles_;
-    if (labelPosition_.get() != LabelPosition::None) {
-        extraMargins_.x += extraMarginsLabel_.x;
-        extraMargins_.z += extraMarginsLabel_.z;
-        extraMargins_.y = std::max(extraMargins_.y, extraMarginsLabel_.y);
-        extraMargins_.w = std::max(extraMargins_.w, extraMarginsLabel_.w);
-    }
-    if (showValue_.get()) {
-        extraMargins_.y = std::max(extraMargins_.y, extraMarginsValues_.y);
-        extraMargins_.w = std::max(extraMargins_.w, extraMarginsValues_.w);
-    }
+    buildTextCache(enabledAxis);
 
-    drawAxis(dims, enabledAxis, extraMargins_);
-    drawLines(dims, enabledAxis, extraMargins_);
-    drawHandles(dims, enabledAxis, extraMargins_);
-    renderText(dims, enabledAxis, extraMargins_);
+    drawAxis(dims, enabledAxis);
+    drawLines(dims);
+    drawHandles(dims, enabledAxis);
+    renderText(dims, enabledAxis);
 
     utilgl::deactivateCurrentTarget();
 }
 
 void ParallelCoordinates::createOrUpdateProperties() {
+    axisVector_.clear();
+    for (auto &p : axisProperties_.getProperties()) {
+        p->setVisible(false);
+    }
     if (dataFrame_.hasData()) {
         auto data = dataFrame_.getData();
         if (!data->getNumberOfRows()) return;
-        axisVector_.clear();
+        if (!data->getNumberOfColumns()) return;
 
-        for (auto &p : axisProperties_.getProperties()) {
-            p->setVisible(false);
-        }
-
-        size_t col = 0;
-
-        // start at 1 to skip index buffer
-        for (size_t i = 1; i < data->getNumberOfColumns(); i++) {
+        for (size_t i = 0; i < data->getNumberOfColumns(); i++) {
             auto c = data->getColumn(i);
+            std::string displayName = c->getHeader();
+            std::string identifier = displayName;  // TODO: remove non alpha num
+
+            util::erase_remove_if(identifier, [](char cc) {
```

Comment:
consider using `util::stripIdentifier()` from `util/utilities.h` instead of doing it yourself

Reply:
fixed